contracts/true-currencies/TrueCurrencyWithPoR.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {TrueCurrency} from "./TrueCurrency.sol";
+import {IChainlinkAggregatorV3} from "../common/interface/IChainlinkAggregatorV3.sol";
+import {IPoRToken} from "../common/interface/IPoRToken.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+/**
+ * @title TrueCurrencyWithPoR
+ * @dev TrueCurrencyPoR is an ERC20 with blacklist & redemption addresses.
+ *  Please see TrueCurrency for the implementation that this contract inherits from.
+ *  This contract implements an additional check against a Proof-of-Reserves feed before
+ *  allowing tokens to be minted.
+ */
+abstract contract TrueCurrencyWithPoR is TrueCurrency, IPoRToken {
+    using SafeMath for uint256;
+
+    uint256 public constant MAX_AGE = 7 days;
+
+    constructor() public {
+        chainReserveHeartbeat = MAX_AGE;
```

Comment:
If we're doing an upgrade of TUSD, then this constructor won't get called. We'll have to tell Techteryx to queue `setChainReserveHeartbeat(7 days)` from the multisig after the upgrade call.

Reply:
As mentioned earlier, I think MAX_AGE is an arbitrary constant, I don't think anybody knows for sure what a really useful/practical MAX_AGE is at the moment. I think it's probably best not to constrain the possible values of `chainReserveHeartbeat` in the code and remove `MAX_AGE` altogether. Please do let me know if you think otherwise.

We need to be able to disable temporarily PoR if Armanino, Chainlink or one of our banking partners' API (most likely) is down for an extended period of time, this will be done be setting `chainReserveFeed` to `0`. We should be able to do it quickly, e.g. in case of a VIP mint, and I think calling `setChainReserveFeed` should be quicker and cheaper than upgrading the contract. Do you agree?

Also, we plan to deploy it on Ethereum and Avalanche, which, I assume, will have different feed addresses, so I think hardcoding `CHAIN_RESERVE_FEED` in the code is probably not a good idea.

Reply:
Having the option to set `chainReserveFeed` to 0 sounds to me like enough reason to keep this as a storage variable! And yeah, I'd overlooked that Ethereum mainnet and Avalanche will have different addresses. Definitely shouldn't hardcode in that case.